### ISSUE TITLE
🚨 🚨 🚨 🚨 THIS REPO IS NOT MAINTAINED: Try this fork instead! 🚨 🚨 🚨 🚨 

### ISSUE THREAD

**Issue by thedonquixotic** [wp-sync-db/wp-sync-db#167]
Github really needs to do something about making more advanced forks easier to find. 

Anyway try this one:

https://github.com/outlandishideas/wp-sync-db

**Comment by adriandmitroca** [comment]
Is there any changelog with listed changes?

**Comment by jcklpe** [comment]
@slang800 No worries! (open source people don't owe users and are being generous with their time :smile:   )I would suggest rather opening up the repo to contributors in the spirit of this idea: http://felixge.de/2013/03/11/the-pull-request-hack.html

Or at least giving repo rights to @corysimmons https://github.com/corysimmons/wp-sync-db or @fahimxyz  https://github.com/fahimxyz/wp-shift-db who have the most advanced forks of the project from my research. Or maybe @outlandishideas would being willing to take up the mantle in a more official capacity. 

The current problem as it is, is that due to the way Github works it's very hard to find which forks are being actively developed. 

@adriandmitroca : There is no changelog to my knowledge but if you look through the commit history of the other branches you can usually see which pull requests they chose to merge in and what issues they were fixing it.

**Comment by jcklpe** [comment]
Also I haven't merged in the changes from the upstream project. I am a designer who is learning to code and I'm still very early in my journey (though I'd love to help once I learn a bit more. I made my first pull request for a javascript plugin a couple of weeks ago! :smiley:  )

**Comment by corysimmons** [comment]
@slang800 Just merge every PR as fast as you can and let god sort it out. >: D

Anyway, I have no interest in maintaining this either. PHP/WordPress is dead asf. Please migrate to better CMS' like Dato or something.

**Comment by jcklpe** [comment]
@corysimmons I don't want to take this thread off topic but Dato isn't open source is it? You have to pay for it like Craft CMS etc right? People have been saying PHP is dead since I first stuck my toe in code but I haven't seen that yet. I feel like I'm constantly playing catch up though 😩

**Comment by corysimmons** [comment]
Dato has free plan, but there are alternatives that everyone should explore instead of keeping WordPress alive. https://headlesscms.org

**Comment by notslang** [comment]
> PHP is dead

I'm willing to echo that sentiment. It's a [messy](https://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/) language and there are very few cases where PHP is truly the best language for the job.

I don't have any stats on this, but in my experience it _seems_ like most developers who started out with PHP reach a certain level of experience, then find a better language to work with, and never go back. I don't see many experienced developers deciding to learn PHP either, so the PHP community ends up with a huge number of relatively unskilled developers, and very few talented devs who are able to contribute to the ecosystem. It's a case of the blind leading the blind.

I can see why giving minor contributors commit access to Node.js projects would work well. There are a ton of easy-to-find, talented people in that ecosystem. In PHP, and especially with WordPress, I don't think the same thing is true.

**Comment by yumyo** [comment]
PHP is dead? Ah! There are stunning Laravel and CodeIgniter web apps that perform as native Mac apps (e.g. Ever tried Koken?). Not to mention that:

> As of December 2017, PHP makes up over 83% of server side languages used on the internet. [link](https://w3techs.com/technologies/details/pl-php/all/all)

and that the latest version (7.1) is an extremely fast, streamlined language with a strong OO focus. Node? Python? Sure! But do not confuse programming languages and programming skills. 

Ok what about WordPress? Beyond the well known stats, WP is undergoing some deep changes and this will keep happening in the future but the most important fact is that **you can just use WP as an headless CMS** almost anyone can or already know how to use (yes, I am thinking about your marketing team)

**Comment by jcklpe** [comment]
Just wanted to follow up. I finally got around to trying the various additional branches of this and some of them did not work but I can confirm that Corey's version worked perfectly:  https://github.com/corysimmons/wp-sync-db

**Comment by notasausage** [comment]
Confirming that [the wp-sync-db fork](https://github.com/corysimmons/wp-sync-db) from @corysimmons is indeed working well. I didn't try the original repo or any of the other forks since you fine folks pointed to Corey's. I'm using this plugin along with a git workflow to version control my local WordPress development that can be pushed and pulled from WP Engine.

Anyone using the [original wp-sync-db-media-files add-on](https://github.com/wp-sync-db/wp-sync-db-media-files), or a fork of it? I haven't tried either yet, but I'm about to install it and try it out.

**Comment by adriandmitroca** [comment]
Media Files plugin works just fine, you don't have to worry about using forks.

**Comment by jcklpe** [comment]
I just tried the Media Files plugin but it didn't work for me. I'm actually not using Corey's branch but a similar one by a developer who said he's going to try and actively maintain and approve the pull requests. So that could be a fluke or that could be me not using it correctly or something idk. Oh an dI'm also using a paid Media Library Plugin so it might be relating my files to the plugin in a non supported way.

**Comment by userabuser** [comment]
@yumyo 
> PHP is dead? Ah! There are stunning Laravel and CodeIgniter web apps that perform as native Mac apps... 

PHP is dead? 😂 Ya I agree with your sentiment... The whole "...is dead" thing is, dead.

I can among other things (like running WP in headless mode with Vue or React front end) run websockets in PHP, CI in PHP, build CLI tools in PHP, embed PHP into cross-platform applications and people would be none the wiser without necessarily having to rely on Node.

PHP plays nice with other languages and you can do a lot with it, but more importantly PHP 7.* and beyond is continuing to improve the language and is undoubtedly moving the language past the point of being a "fractal of bad design", just look at the [RFCs](https://wiki.php.net/rfc) past, present and proposed. 

Unfortunately people who tout that PHP is dead really don't understand the ecosystem, industry and community. The same applies to when I hear people say Ruby is dead... Although in decline, it's far from dead, yet.

Anyway in 5 years everything is going to be JavaScript 🤣

**Comment by corysimmons** [comment]
>Anyway in 5 years everything is going to be JavaScript 🤣

<img width="200px" src="https://user-images.githubusercontent.com/224636/50314695-e0c5e700-047d-11e9-9c63-27a4da78e152.png">

5 years **ago** everything migrated to React and Node lambdas. 🧐

**Comment by userabuser** [comment]
@corysimmons Indeed you are correct... This was even foretold in the book of Revelation, chapter 1, verse 8:

> I am the Alpha and the Omega, says the Lord ~~God~~ React,
> who is and who was and who is to come, the Almighty

**Comment by JanSichula** [comment]
Hello guys, at GitHub, I would suggest to avoid flame wars as far as preferred programming technologies go and whether PHP or any language is already dead or not. The GitHub ISSUES tab is descriptive by itself from it's very title. We should stick to issues and solutions for a particular project. That is about it.

@userabuser I have sent you a message through Hangouts service.

**Comment by djboris88** [comment]
Why is this thread pointing to the repositories not updated for more than a year? The official version (this repo) is working fine for me. 

I also don't understand all the death-wishes for the PHP and WP. I've seen many bad examples of the WP and PHP projects, but I've also seen a many beautiful ones. I'm sure the same applies to JS or any other language/framework. I personally love building WP projects, but I've set my standards as high as possible. It makes me really sad to see your opinions on WP or PHP.

I love it because it is quick to develop complex Apps, and nowadays you can really do it with high quality and performance. Also, clients love it for the low cost hosting solutions and maintenance. I do OOP and DRY as much as possible, using .twig (Timber) templates, custom theme autoloader Classes, with caching in mind...

**Comment by cyphix333** [comment]
Yep, this repo itself seems just as active as any of the other repos listedl the one by @corysimmons hasn't had a commit since 2017 for example.

@slang800 Did no official maintainer take this over? Is this pretty much dead?

**Comment by thatnerdjosh** [comment]
@adriandmitroca have you used the media sync plugin on wordpress 5.2.3? It doesn't seem to work for me, it is activated and checked but it doesn't pull the media files

Temporary workaround:
```
scp -r <server>:/home/customer/www/public_html/wp-content/uploads/* wp-content/uploads/
```

**Comment by thatnerdjosh** [comment]
@cyphix333 I can take over maintenance of this, I will work on porting the code this week so the latest version can be packaged. I've been looking for an open source project to maintain

EDIT:
  This is a bigger job than I had anticipated, there are a ton of files now in the source repo as it looks like he rearchitected it, so I am working on figuring out what I need to remove in order to bring in just the functionality for this repo. I will keep you guys posted :D

**Comment by notslang** [comment]
@NerdsvilleCEO - the only changes I've made are renaming "WP Migrate DB Pro" to "WP Sync DB" (avoiding a trademark issue that was built into the code), removing license code, and a little reformating. No rearchitecting. If you look [in this branch](https://github.com/wp-sync-db/wp-sync-db/commits/delicious-brains) you will see the code exactly as it was released by delicious brains.

**Comment by thatnerdjosh** [comment]
@notslang, not you, the upstream... Take a look at the latest code... Example: https://github.com/deliciousbrains/wp-migrate-db/blob/master/class/Container.php, also look at all the files in the class directory

I'm just wrapping my head around it all and pulling out the features that you have in other repositories

**Comment by notslang** [comment]
@cyphix333 yeah, nobody has taken this over. I've had several people express interest, but nobody has worked on it.

I haven't worked on WordPress or PHP in several years, so I've had no reason to touch this.

**Comment by notslang** [comment]
@NerdsvilleCEO Oh, sorry. That's not really the upstream. I thought that was what they based their "WP Migrate DB Pro" codebase on when I created this repo, so I forked from that and added the new code on top of that repo. However, it turns out it's a totally different codebase. I should probably just unfork it since it's kinda confusing.

https://github.com/deliciousbrains/wp-migrate-db is only a spiritual ancestor, not a real basis for this code.

**Comment by thatnerdjosh** [comment]
Ok so where do I find the latest? Also it seems they contribute to that repository still, are you certain that's not the correct one?

Edit: should I download their plugin and edit those files instead?

**Comment by notslang** [comment]
> Also it seems they contribute to that repository still, are you certain that's not the correct one?

They are still making commits to https://github.com/deliciousbrains/wp-migrate-db because it's a separate project. WP Sync DB is a free version of WP Migrate DB Pro. It is not actually related to WP Migrate DB.

> Where do I find the latest

You can look in the delicious-brains branch for a slightly more recent one, if you want to go through the code version by version and add in the changes. Or you could download the latest version from delicious brains if you want the absolute latest version. If you go that route, you're going to need an API key to make the request to `https://api.deliciousbrains.com/?wc-api=delicious-brains&request=download&licence_key=<api-key>&slug=wp-migrate-db-pro&site_url=example.com`. You can either buy the plugin to get that license key, or just find one on GitHub with a clever search query if you're cheap.

**Comment by thatnerdjosh** [comment]
Thanks, I haven't forgotten about this, just started a new contract but hoping to get to this soon! Main blocker on this is buying a license for the software so I can pull it down :-) would prefer to buy it outright to avoid any issues

**Comment by frontenddevguy** [comment]
It seems that none of the forks are currently working. This is a shame as I also cannot yet manage to get Wordmove db push/pull functioning on my system. It makes developing on WordPress quite difficult :(

**Comment by jcklpe** [comment]
@frontenddevguy I tried this one just a couple of weeks ago and it worked fine: https://github.com/cloudverve/wp-sync-db

I just gitcloned it into my plugins folder and used it. The media sync plugin doesn't work with it to my knowledge but that stuff can be dealt pretty easily with scp/sftp-clients

**Comment by calvinwinter** [comment]
If anyone is looking for a working version of please check out my fork of the current version of WP Migrate DB Pro from Delicious Brains https://github.com/calvinwinter/wp-database-tools

**Comment by ermincelikovic** [comment]
> If anyone is looking for a working version of please check out my fork of the current version of WP Migrate DB Pro from Delicious Brains https://github.com/calvinwinter/wp-database-tools

This looks great @calvinwinter!
Does it work with the media sync add-on?
https://github.com/wp-sync-db/wp-sync-db-media-files

**Comment by bdlowery** [comment]
> > If anyone is looking for a working version of please check out my fork of the current version of WP Migrate DB Pro from Delicious Brains https://github.com/calvinwinter/wp-database-tools
> 
> This looks great @calvinwinter! Does it work with the media sync add-on? https://github.com/wp-sync-db/wp-sync-db-media-files

No, it does not. I just checked.

**Comment by jsongerber** [comment]
Here is an other fork of this plugin I made:  
https://github.com/jsongerber/wp-sync-db  
https://github.com/jsongerber/wp-sync-db-media-files  

I don't plan to add new features or get features from Delicious Brain, I just fixed the composer file and a deprecation notices to make it installable and usable.  

I use it a lot, so I should be fixing things as they break.

**Comment by bdlowery** [comment]
> Here is an other fork of this plugin I made: https://github.com/jsongerber/wp-sync-db https://github.com/jsongerber/wp-sync-db-media-files
> 
> I don't plan to add new features or get features from Delicious Brain, I just fixed the composer file and a deprecation notices to make it installable and usable.
> 
> I use it a lot, so I should be fixing things as they break.

legend! thank you.
